helix-core/src/main/java/org/apache/helix/HelixManagerProperty.java

```diff
@@ -56,13 +60,15 @@ public HelixManagerProperty(Properties helixManagerProperties, CloudConfig cloud
 
   private HelixManagerProperty(String version, long healthReportLatency,
       HelixCloudProperty helixCloudProperty,
-      RealmAwareZkClient.RealmAwareZkConnectionConfig zkConnectionConfig,
-      RealmAwareZkClient.RealmAwareZkClientConfig zkClientConfig) {
+      RealmAwareZkClient.RealmAwareZkConnectionConfig realmAwareZkConnectionConfig,
+      RealmAwareZkClient.RealmAwareZkClientConfig zkClientConfig,
+      HelixZkClient.ZkConnectionConfig zkConnectionConfig) {
```

Comment:
if i read your previous comment, you want either realmAwareZkConnection or regular ZkConnection, so can we not have 2 distinct constructors and not break any existing users?